### PR TITLE
Add streaming speech synthesis support.

### PR DESCRIPTION
**Pull request by shaper**
Things still to do in the future:
- use npm module or shell command to play, rather than user having to do it themselves
- update streaming demo to play audio as it comes in
- write tests for streaming synthesis
- add error callback for streaming synthesis
- update js docs to describe streaming

### CODE REVIEW

File: speech.js [hunk 1]
```diff
@@ -1,5 +1,66 @@
...
+import WebSocket from 'ws';
+
+const DEBUG = false;
+const logDebug = (message) => {
+  if (DEBUG) {
+    console.debug(message);
+  }
+};
+
+class StreamingSynthesisConnection {
```
Comment (zachoverflow) [inline speech.js:12]: can we make this less callback heavy?

File: speech.js [hunk 2]
```diff
@@ -1,5 +1,105 @@
...
+  push(message) {
+    if (this._resolvers.length) {
+      const resolve = this._resolvers.shift();
+      resolve(message);
+    } else {
+      this._messages.push(message);
+    }
+  }
+
+  async getNextMessage() {
```
Comment (zachoverflow) [inline speech.js:35]: next()?

since it's a message queue and message is already in the name

**Review by zachoverflow — APPROVED** [review]
nice! feels much more ergonomic now

File: speech.js (hunk 2 above)
Comment (shaper) [reply speech.js]: Done.

**Pull request merged**